### PR TITLE
Add Twilio SMS connector

### LINKED ISSUE #1490: Creating A SMS Connector
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; I hope your staying safe! A SMS connector could really benefit this platform, as there are so many existing applications for text bots. I think we could do this using the web server functionality, and the Twilio API. 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I have started this in #1488, and I'm not finished but I've opened a draft pull request if anyone has feedback. I know parts of it don't even work, so if anyone wants to take a crack at it, I would be grateful

### PR DESCRIPTION
**Pull request by danielballan**
# Description

Closes #1490 
Supersedes #1488 

This builds on @Gideon357's commits in #1488 with the following changes:

* Addressed @jacobtomlinson's review comments left on the original PR.
* Confirmed that there is no programmatic API for configuring the webhook route in Twilio, left a comment with a link.
* Fixed some things to get it working.
* Moved all synchronous calls into an executor. The Twilio Python API does not support async, per [this official post](https://www.twilio.com/blog/twilio-python-helper-library-async), and recommends several approaches. This is the most straightfoward one, IMO.
* Implemented security, such that requests not correctly signed by Twilio are rejected with `HTTP 403`.

This works for me interactively. I have branched `v0.19.0` for now just to avoid getting confused by changes since. Additional work may be need to make it work on `master`. Any hints about how best practices have evolved since `v0.19.0` would be welcome. :- )

TO DO:

- [ ] Rebase on `master` and make any necessary changes
- [ ] Revisit docs
- [ ] More tests

## Status
READY | **UNDER DEVELOPMENT** | ON HOLD

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Needs tests

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/connector/sms/__init__.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            None,
+            Client,
+            self.config["account_sid"],
+            self.config["auth_token"]
+        )
+        # This route must be configured manually in the Twilio web GUI.
+        # There is no API for configuring it.
+        # Watch this SO post for any developments.
+        # Last updated by Twilio as of August 2019.
+        # https://stackoverflow.com/a/57420558/1221924
```
Comment (jacobtomlinson) [inline opsdroid/connector/sms/__init__.py:55]: It would be good to add this to the docs, and perhaps log it too?

**Comment by danielballan** [comment]
Some weekend I still circle back to this. I am running a hobby project on the branch in whatever the hobby equivalent of “production” is and it works so it’s worth finishing off to a point that it can be merged.

**Comment by jacobtomlinson** [comment]
@danielballan it would be great to see this in a state that we can merge.

**Comment by danielballan** [comment]
Agreed! Around the time I opened this the test harness was in a state of flux. Is there a connector in particular that I should refer to for writing test, or are all of them taking a similar approach and a similar standard of rigor?

**Comment by jacobtomlinson** [comment]
Yeah things have settled a bit now. The GitHub connector is probably the gold standard right now, and we generally insist on 100% coverage.

**Comment by danielballan** [comment]
Thanks @jacobtomlinson !

**Pull request closed without merging**